cli/collection_patch.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+func MakeCollectionPatchCommand() *cobra.Command {
+	var patchFile string
+	var cmd = &cobra.Command{
+		Use:   "patch [patch]",
+		Short: "Patch existing collection descriptions",
+		Long: `Patch existing collection descriptions.
+
+Uses JSON Patch to modify collection descriptions.
+
+Example: patch from an argument string:
+  defradb client collection patch '[{ "op": "add", "path": "...", "value": {...} }]'
+
+Example: patch from file:
+  defradb client collection patch -f patch.json
+
+Example: patch from stdin:
+  cat patch.json | defradb client collection patch -
+
+To learn more about the DefraDB GraphQL Schema Language, refer to https://docs.source.network.`,
+		Args: cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			store := mustGetContextStore(cmd)
+
+			var patch string
+			switch {
+			case patchFile != "":
+				data, err := os.ReadFile(patchFile)
+				if err != nil {
+					return err
+				}
+				patch = string(data)
+			case len(args) > 0 && args[0] == "-":
+				data, err := io.ReadAll(cmd.InOrStdin())
+				if err != nil {
+					return err
+				}
+				patch = string(data)
+			case len(args) == 1:
+				patch = args[0]
+			default:
+				return fmt.Errorf("patch cannot be empty")
+			}
+
+			return store.PatchCollection(cmd.Context(), patch)
+		},
+	}
+	cmd.Flags().StringVarP(&patchFile, "patch-file", "p", "", "File to load a patch from")
```

Comment:
Don't all other file consumption commands use `--file |-f` why do we want to have a different flag name for this arg (does it clash with `-f` somewhere?)

Reply:
`PatchSchema` uses `p` because it it takes two kinds of files.  I think it is more important to be consistent with `PatchSchema` than anywhere else here.

Reply:
How does it take two files? I see 1 so far with -p

Reply:
`PatchSchema` takes two files, I assume you are looking at `PatchCollection`

Reply:
Maybe we should change the flag for patch schema to also be -f and make it consistent throughout. The lens file can stay with -t or maybe -l would make more sense.

Reply:
`f` and `file` is ambiguous for a command that takes two different files.  I do not like that.

Reply:
@nasdf your opinion would be appreciated here when you have a chance.

Reply:
I think its fine to keep it `-p` for consistency with the other patch method. Using `-f` when there are multiple files is worse in my opinion.